Renamed a TensorListMetaData property. Cleaned up a test

izdeby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46664 Move device guard from MultiTensorApply.cuh
* #46663 Remove .pyi from /_multi_tensor
* **#46662 Renamed a TensorListMetaData property. Cleaned up a test**
* #46661 Refactor arguments checks for foraech APIs

Differential Revision: [D24453346](https://our.internmc.facebook.com/intern/diff/D24453346)

Commit:
Renamed a TensorListMetaData property. Cleaned up a test

[ghstack-poisoned]

Commit:
Update on "Renamed a TensorListMetaData property. Cleaned up a test"




[ghstack-poisoned]

Commit:
Update on "Renamed a TensorListMetaData property. Cleaned up a test"


Differential Revision: [D24453346](https://our.internmc.facebook.com/intern/diff/D24453346)

[ghstack-poisoned]

Commit:
Update on "Renamed a TensorListMetaData property. Cleaned up a test"


Differential Revision: [D24453346](https://our.internmc.facebook.com/intern/diff/D24453346)

[ghstack-poisoned]

zou3519 commented on test/test_foreach.py:
```diff
@@ -703,56 +703,57 @@ def test_bin_op_scalar_with_different_tensor_dtypes(self, device):
+            tensors2 = []
+
+            # Empty lists
+            with self.assertRaises(RuntimeError):
+                bin_op(tensors1, tensors2)
+            with self.assertRaises(RuntimeError):
+                bin_op_(tensors1, tensors2)
+
```
nit: prefer AssertRaisesRegex so that we can check that the error message is sane

zou3519 commented on test/test_foreach.py:
```diff
@@ -703,56 +703,57 @@ def test_bin_op_scalar_with_different_tensor_dtypes(self, device):
+                with self.assertRaisesRegex(RuntimeError, "Expected all tensors to be on the same device"):
+                    bin_op_([tensor1], [tensor2])
+
+            # Coresponding tensors with different sizes
```
nit: "coresponding" -> "corresponding"

izdeby replied:
Done

zou3519 commented:
btw, tests are failing with:
```
Oct 23 20:41:06 ====================================================================== 
Oct 23 20:41:06 FAIL [0.016s]: test_bin_op_list_error_cases_cpu (__main__.TestForeachCPU) 
Oct 23 20:41:06 ---------------------------------------------------------------------- 
Oct 23 20:41:06 RuntimeError: Scalars list must have at least one value. 
Oct 23 20:41:06  
Oct 23 20:41:06 During handling of the above exception, another exception occurred: 
Oct 23 20:41:06  
Oct 23 20:41:06 Traceback (most recent call last): 
Oct 23 20:41:06   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 274, in instantiated_test 
Oct 23 20:41:06     result = test_fn(self, *args) 
Oct 23 20:41:06   File "test_foreach.py", line 720, in test_bin_op_list_error_cases 
Oct 23 20:41:06     bin_op(tensors1, tensors2) 
```

izdeby committed:
Update on "Renamed a TensorListMetaData property. Cleaned up a test"


Differential Revision: [D24453346](https://our.internmc.facebook.com/intern/diff/D24453346)

[ghstack-poisoned]

ngimel commented on test/test_foreach.py:
```diff
@@ -799,33 +800,6 @@ def test_div_list(self, device, dtype):
-                bin_op(tensors1, tensors2)
-
-            # Lists have different amount of tensors
-            tensors2.append(torch.tensor([1], device=device))
```
I don't see this test anymore - did I miss it?

izdeby replied:
These test cases were duplicated. You can see them in `test_bin_op_list_error_cases`

ngimel replied:
Oh thanks, I'm blind.

Commit:
Update on "Renamed a TensorListMetaData property. Cleaned up a test"


Differential Revision: [D24453346](https://our.internmc.facebook.com/intern/diff/D24453346)

[ghstack-poisoned]

Commit:
Update on "Renamed a TensorListMetaData property. Cleaned up a test"


Differential Revision: [D24453346](https://our.internmc.facebook.com/intern/diff/D24453346)

[ghstack-poisoned]
